modules/Bio/EnsEMBL/Variation/BaseTranscriptVariation.pm

```diff
@@ -218,12 +276,67 @@ sub cds_start {
 =cut
 
 sub cds_end {
-    my ($self, $cds_end) = @_;
-    
+    my ($self, $cds_end, $toshift) = @_;
+
+    my $shifting_offset = defined($toshift) ? $toshift : 0;
```

Comment:
This is somewhere I'd use:

my ($self, $cds_end, $shifting_offset ) = @_;
$shifting_offset ||=0;

to avoid creating additional variable names to keep track of.